code/modules/storage/storage.dm

```diff
@@ -0,0 +1,413 @@
+/datum/storage/proc/storage_item_after_attack(atom/target, mob/user, reach)
+	// if item is stored, drop storage and take it out
+	if (target in src.get_contents())
+		user.drop_item()
+		src.transfer_stored_item(target, get_turf(src.linked_item), user = user)
+		SPAWN(1 DECI SECOND)
+			target.Attackhand(user)
+		return
+	// attempt to load item into storage if you have a free hand
+	else if (isitem(target) && !istype(target, /obj/item/storage))
+		var/obj/O = target
+		if (O.anchored)
+			return
+		if (!can_reach(user, target))
+			return
+		user.swap_hand()
+		if (user.equipped() == null)
+			target.Attackhand(user)
+			if (target in user.equipped_list())
+				src.add_contents(target, user)
+		else
+			boutput(user, "<span class='notice'>Your hands are full!</span>")
+		user.swap_hand()
+
+/// storage item is dropped
+/datum/storage/proc/storage_item_on_drop(atom/source, mob/user)
+	src.hud?.update(user)
+
+/// when reaching inside the storage item, check for traps
+/datum/storage/proc/mousetrap_check(mob/user)
+	if (!ishuman(user) || user.stat)
+		return FALSE
+	for (var/obj/item/mousetrap/MT in src.get_contents())
+		if (MT.armed)
+			user.visible_message("<span class='alert'><B>[user] reaches into \the [src.linked_item.name] and sets off a mousetrap!</B></span>",\
+				"<span class='alert'><B>You reach into \the [src.linked_item.name], but there was a live mousetrap in there!</B></span>")
+			MT.triggered(user, user.hand ? "l_hand" : "r_hand")
+			return TRUE
+	for (var/obj/item/mine/M in src.get_contents())
+		if (M.armed && M.used_up != TRUE)
+			user.visible_message("<span class='alert'><B>[user] reaches into \the [src.linked_item.name] and sets off a [M.name]!</B></span>",\
+				"<span class='alert'><B>You reach into \the [src.linked_item.name], but there was a live [M.name] in there!</B></span>")
+			M.triggered(user)
+			return TRUE
+
+/// show storage contents
+/datum/storage/proc/show_hud(mob/user)
+	user.s_active = src.hud
+	src.hud.update(user)
+	user.attach_hud(src.hud)
+
+/// hide storage
+/datum/storage/proc/hide_hud(mob/user)
+	if (user.s_active == src.hud)
+		user.s_active = null
+		user.detach_hud(src.hud)
+
+// ----------------- PUBLIC PROCS ----------------------
+
+/// check if the storage can hold an item or not
+/datum/storage/proc/check_can_hold(obj/item/W)
+	if (!W)
+		return STORAGE_CANT_HOLD
+
+	if (W.cant_drop)
+		return STORAGE_WONT_FIT
+
+	// if can_hold is defined, check against that
+	if (length(src.can_hold))
+		// early skip if weight class is allowed
+		if (src.check_wclass && W.w_class <= src.max_wclass)
+			return STORAGE_CAN_HOLD
+		for (var/type in src.can_hold)
+			if (ispath(type) && istype(W, type))
+				return STORAGE_CAN_HOLD
+		for (var/type in src.can_hold_exact)
+			if (ispath(type) && W.type == type)
+				return STORAGE_CAN_HOLD
+		return STORAGE_CANT_HOLD
+
+	else if (W.w_class > src.max_wclass)
+		return STORAGE_WONT_FIT
+
+	if (src.is_full())
+		return STORAGE_IS_FULL
```

Comment:
Tiny nitpick- imo this should go near the top of the proc with the rest of the quick `if condition return CODE` blocks

Reply:
Initial idea was just for the message given to the user, so that if they try to fit something larger than can be held in a backpack that is full, they will get the "the backpack can't hold that" before the "the backpack is full" message. Will still change if wanted

Reply:
Makes sense